TitleEdit/PluginServices/Lobby/LobbyService.Housing.cs

```diff
@@ -0,0 +1,156 @@
+        SafeMemory.Write(territoryTypeAddress, territory);
+
+        try
+        {
+            var housingManager = HousingManager.Instance();
+            if (housingManager == null) return;
+
+            initializeHousingLayout(housingManager, territory);
+        }
+        catch (Exception ex)
+        {
+            Services.Log.Error(ex, "Error initializing housing layout");
+        }
+    }
+
+    private void SetupHousing(LocationModel model)
+    {
+        InitializeHousingLayout(model);
+        
+        if (model.SaveLayout == false || model.SaveHousing == false) return;
+        LoadEstate(model);
+        LoadPlots(model);
+        SpawnFurniture(model);
+    }
+
+    private void LoadEstate(LocationModel model)
+    {
+        if (model.Estate == null) return;
+
+        var housingManager = HousingManager.Instance();
+        if (housingManager->IndoorTerritory == null) return;
+
+        var layoutManager = LayoutWorld.Instance()->ActiveLayout;
+        if (layoutManager == null) return;
+
+        for (var floorIndex = 0; floorIndex < 2; floorIndex++)
+        {
+            var floor = model.Estate.Floors[floorIndex];
+            setInteriorFixture(floorIndex, 0, 0, floor.Part0, byte.MaxValue);
+            setInteriorFixture(floorIndex, 1, 0, floor.Part1, byte.MaxValue);
+            setInteriorFixture(floorIndex, 2, 0, floor.Part2, byte.MaxValue);
+            setInteriorFixture(floorIndex, 3, 0, floor.Part3, byte.MaxValue);
+            setInteriorFixture(floorIndex, 4, 0, floor.Part4, byte.MaxValue);
+        }
+
+        if (layoutManager->IndoorAreaData != null)
+        {
+            layoutManager->IndoorAreaData->LightLevel = model.Estate.LightLevel;
+        }
+    }
+
+    private void SetPlot(HousingPlotModel plotModel)
+    {
+        if (plotModel.Plot >= 60) return;
+        
+        var housingManager = HousingManager.Instance();
+        if (housingManager == null) return;
+        if (housingManager->OutdoorTerritory == null) return;
+        var layoutManager = LayoutWorld.Instance()->ActiveLayout;
+        if (layoutManager == null) return;
+        if (layoutManager->OutdoorAreaData == null) return;
+        
+        for (byte fixtureIndex = 0; fixtureIndex < plotModel.Fixtures.Length && fixtureIndex < 8; fixtureIndex++)
+        {
+            layoutManager->OutdoorAreaData->SetFixture(plotModel.Plot + 1U, fixtureIndex, plotModel.Fixtures[fixtureIndex].FixtureId);
+            layoutManager->OutdoorAreaData->SetFixtureStain(plotModel.Plot + 1U, fixtureIndex, plotModel.Fixtures[fixtureIndex].StainId);
+        }
+    }
+    
+    private void LoadPlots(LocationModel model)
+    {
+        if (model.Plots is not { Count: > 0 }) return;
+        foreach (var plotModel in model.Plots)
+        {
+            Services.Framework.RunOnTick(() => {
+                SetPlot(plotModel);
+            }, delayTicks: plotModel.Plot * 2);
```

Comment:
What's the point of delaying loading like this? To prevent a hitch? One problem I see is if the scene changes while some of these are still running. They would need to be cancelled on a new scene load  when `CreateSceneDetour` is called.

Reply:
theres no hitch, but it resolved issues with random houses not spawning in to delay it. I'll add a cancellation token

Reply:
Took a look at it and found a method that is called during layout initialization that is responsible for doing some initialization with the OutdoorAreaLayoutData structs, setting the plots right after that seems to fix the issue of houses not appearing and also makes the pop-in much less noticable. I'll merge your PR, add my changes and push out the update, and hope my changes don't burn anything down :)